src/shogun/multiclass/CoverTreeKNNSolver.cpp

```diff
@@ -58,7 +57,7 @@ CMulticlassLabels* CCoverTreeKNNSolver::classify_objects(CDistance* distance, co
 	v_array< v_array< CJLCoverTreePoint > > res;
 	k_nearest_neighbor(top, top_query, res, m_k);
 
-#ifdef DEBUG_KNN
+#ifdef SG_DEBUG
```

Comment:
Oh, now I understand. Does it works like:

>
	SG_DEBUG("\nJL Results:\n")
	for ( int32_t i = 0 ; i < res.index ; ++i )
	{
		for ( int32_t j = 0 ; j < res[i].index ; ++j )
		{
			SG_DEBUG("%d ", res[i][j].m_index);
		}
		SG_DEBUG("\n");
	}
	SG_DEBUG("\n")
>

Is that mean it will run this for lop every time? even it wouldn't output anything if we aren't debugging? Maybe we should keep the `DEBUG_KNN` and just use SG_DEBUG instead of SG_PRINT?